src/mpi/pt2pt/recv.c

```diff
@@ -145,6 +145,19 @@ int MPI_Recv(void *buf, int count, MPI_Datatype datatype, int source, int tag,
         goto fn_exit;
     }
 
+    if (unlikely(MPIR_CVAR_ENABLE_FT &&
```

Comment:
Correct. That's why they were put there to begin with.

Reply:
@danghvu these patches are just optimizations, right? Can we drop them for now? There should be a way to disable FT at configure time to make these branches go away in the compiled library.